src/gmtdefaults.1
.TH GMTDEFAULTS 1 "Feb 27 2014" "GMT 4.5.13 (SVN)" "Generic Mapping Tools"
.SH NAME
gmtdefaults \- To list current \fBGMT\fP defaults
.SH SYNOPSIS
\fBgmtdefaults\fP \fB\-D\fP[\fBu\fP|\fBs\fP] | \fB\-L\fP
.SH DESCRIPTION
\fBgmtdefaults\fP lists the 
\fBGMT\fP parameter defaults if the option \fB\-D\fP is used.  There are three ways to change some of the settings:
(1) Use the command \fBgmtset\fP, (2) use any texteditor to edit the file \.gmtdefaults4 in your home, ~/.gmt
or current directory (if you do not have this file, run 
\fBgmtdefaults\fP -D > ~/.gmtdefaults4 to get one with the system default settings), or (3) override any parameter
by specifying one or more \fB\-\-PARAMETER\fP=\fIvalue\fP statements on the commandline of any \fBGMT\fP command
(\fBPARAMETER\fP and \fBVALUE\fP are any combination listed below).  The
first two options are permanent changes until explicitly changed back, while the last option is ephemeral
and only applies to the single \fBGMT\fP command that received the override.
\fBGMT\fP can provide default values in US or SI units.  This choice is determined by the contents of the
gmt_setup.conf file in \fBGMT\fP's share directory.\"'
.TP
\fB\-D\fP
Print the system \fBGMT\fP defaults to standard output.  Append \fBu\fP for US
defaults or \fBs\fP for SI defaults. [\fB\-D\fP alone gives current choice in gmt_setup.conf].
.TP
\fB\-L\fP
Print the user's currently active defaults to standard output.\"'
.br
.sp
Your currently active defaults come from the \.gmtdefaults4 file in the 
current working directory, if present; else from the \.gmtdefaults4 file 
in your home directory, if present; else from the file ~/.gmt/.gmtdefaults4, if present;
else from the system defaults set at the time \fBGMT\fP was compiled.
.SH \fBGMT\fP PARAMETERS
The following is a list of the parameters that are user-definable in \fBGMT\fP.  The
parameter names are always given in UPPER CASE.  The parameter values are
case-insensitive unless otherwise noted.  The
system defaults are given in brackets [ for SI (and US) ].  Those marked \fB*\fP can be set on the command
line as well (the corresponding option is given in parentheses).  Note that default distances and
lengths below are given in both cm or inch; the chosen default depends on your choice of default
unit (see \fBMEASURE_UNIT\fP).  You can explicitly specify the unit used for distances and
lengths by appending \fBc\fP (cm), \fBi\fP (inch), \fBm\fP (meter), or \fBp\fP (points).
When no unit is indicated the value will be assumed to be in the unit set by \fBMEASURE_UNIT\fP.
Note that the printer resolution \fBDOTS_PR_INCH\fP is always the number of dots or pixels per inch.
Several parameters take only TRUE or FALSE.
.TP
.B ANNOT_FONT_PRIMARY
Font used for upper annotations, etc. [Helvetica].  Specify either the font number or
the font name (case sensitive!).  The 35 available fonts are:
.sp
.br
0	Helvetica
.br
1	Helvetica-Bold
.br
2	Helvetica-Oblique
.br
3	Helvetica-BoldOblique
.br
4	Times-Roman
.br
5	Times-Bold
.br
6	Times-Italic
.br
7	Times-BoldItalic
.br
8	Courier
.br
9	Courier-Bold
.br
10	Courier-Oblique
.br
11	Courier-BoldOblique
.br
12	Symbol
.br
13	AvantGarde-Book
.br
14	AvantGarde-BookOblique
.br
15	AvantGarde-Demi
.br
16	AvantGarde-DemiOblique
.br
17	Bookman-Demi
.br
18	Bookman-DemiItalic
.br
19	Bookman-Light
.br
20	Bookman-LightItalic
.br
21	Helvetica-Narrow
.br
22	Helvetica-Narrow-Bold
.br
23	Helvetica-Narrow-Oblique
.br
24	Helvetica-Narrow-BoldOblique
.br
25	NewCenturySchlbk-Roman
.br
26	NewCenturySchlbk-Italic
.br
27	NewCenturySchlbk-Bold
.br
28	NewCenturySchlbk-BoldItalic
.br
29	Palatino-Roman
.br
30	Palatino-Italic
.br
31	Palatino-Bold
.br
32	Palatino-BoldItalic
.br
33	ZapfChancery-MediumItalic
.br
34	ZapfDingbats
.TP
.B ANNOT_FONT_SIZE_PRIMARY
Font size (> 0) for map annotations  [14p].
.TP
.B ANNOT_FONT_SECONDARY
Font to use for time axis secondary annotations.  See \fBANNOT_FONT_PRIMARY\fP for available fonts  [Helvetica].
.TP
.B ANNOT_FONT_SIZE_SECONDARY
Font size (> 0) for time axis secondary annotations  [16p].
.TP
.B ANNOT_MIN_ANGLE
If the angle between the map boundary and the annotation baseline is less than this minimum value (in degrees), the
annotation is not plotted (this may occur for certain oblique projections.)  Give a value in the
range 0\-90. [20]
.TP
.B ANNOT_MIN_SPACING
If an annotation would be plotted less than this minimum distance from its closest neighbor, the
annotation is not plotted (this may occur for certain oblique projections.) [0]
.TP
.B ANNOT_OFFSET_PRIMARY
Distance from end of tickmark to start of annotation [0.2\fBc\fP (or 0.075\fBi\fP)].  A negative
offset will place the annotation inside the map border.
.TP
.B ANNOT_OFFSET_SECONDARY
Distance from base of primary annotation to the top of the secondary annotation [0.2\fBc\fP (or 0.075\fBi\fP)] (Only
applies to time axes with both primary and secondary annotations).
.TP
.B BASEMAP_AXES
Sets which axes to draw and annotate.  Case sensitive: Upper case means both draw and annotate, lower case
means draw axis only.  [WESN].
.TP
.B BASEMAP_FRAME_RGB
Color used to draw map boundaries and annotations.
Give a red/green/blue triplet, with each element in the 0\-255 range.
Prepend '+' to replicate this color to the tick-, grid-, and frame-pens.
[0/0/0] (black).
.TP
.B BASEMAP_TYPE
Choose between inside, graph, plain and fancy (thick boundary, alternating black/white frame; append + for rounded corners)  [fancy].
For some map projections (e.g., Oblique Mercator), plain is the only option even if fancy is set as default.
In general, fancy only applies to situations where the projected x and y directions parallel the lon and lat
directions (e.g., rectangular projections, polar projections).  For situations where all boundary ticks and annotations
must be inside the maps (e.g., for preparing geotiffs), chose inside. Finally, graph is used for linear projections only
and will extend the axis by 7.5% and add arrow heads.
.TP
.B CHAR_ENCODING
Names the eight bit character set being used for text in files and in command line parameters. This allows \fBGMT\fP
to ensure that the \fIPostScript\fP output generates the correct characters on the plot..  Choose from Standard, Standard+, ISOLatin1, ISOLatin1+,
and ISO-8859-x (where x is in the ranges 1-10 or 13-15).  See Appendix F for details [ISOLatin1+ (or Standard+)].
.TP
.B COLOR_BACKGROUND
Color used for the background of images (i.e., when z < lowest colortable entry).
Give a red/green/blue triplet, with each element in the 0\-255 range. [0/0/0] (black)
.TP
.B COLOR_FOREGROUND
Color used for the foreground of images (i.e., when z > highest colortable entry).
Give a red/green/blue triplet, with each element in the 0\-255 range. [255/255/255] (white)
.TP
.B COLOR_IMAGE
Selects which operator to use when rendering bit-mapped color images.  Due to the
lack of the colorimage operator in some \fIPostScript\fP implementations, as well as
some \fIPostScript\fP editors inability to handle color gradations, \fBGMT\fP offers
two different options:
.br
.sp
	adobe (Adobe's colorimage definition)  [Default].\"'
.br
	tiles (Plot image as many individual rectangles).
.br
.TP
.B COLOR_MODEL
Selects if color palette files contain RGB values (r,g,b in 0-255 range), HSV values
(h in 0-360, s,v in 0-1 range), or CMYK values (c,m,y,k in 0-1 range).  A \fBCOLOR_MODEL\fP setting in the color palette file
will override this setting.  Internally, color interpolation takes place directly
on the RGB values which can give unexpected hues, whereas interpolation directly on the
HSV values better preserves the hues.  Prepend the prefix "+" to force interpolation in
the selected color system (does not apply to the CMYK system). For this additional option, the defaults take
precedence over the color palette file  [rgb].
.TP
.B COLOR_NAN
Color used for the non-defined areas of images (i.e., where z == NaN).
Give a red/green/blue triplet, with each element in the 0\-255 range. [128/128/128] (gray)
.TP
.B D_FORMAT
Output format (C language printf syntax) to be used when printing double precision floating point numbers.
For geographic coordinates, see \fBOUTPUT_DEGREE_FORMAT\fP. [%.12g].
.TP
.B DEGREE_SYMBOL
Determines what symbol is used to plot the degree symbol on geographic map annotations.  Choose between
ring, degree, colon, or none [ring].
.TP
.B DOTS_PR_INCH
Resolution of the plotting device (dpi).  Note that in order to be as compact as possible,
\fBGMT\fP \fIPostScript\fP output uses integer formats only so the resolution should be
set depending on what output device you are using.  E.g, using 300 and sending the output to
a Linotype 300 phototypesetter (2470 dpi) will not take advantage of the extra resolution (i.e.,
positioning on the page and line thicknesses are still only done in steps of 1/300 inch; of course, text will look smoother)  [300].
.TP
.B ELLIPSOID
The (case sensitive) name of the ellipsoid used for the map projections  [WGS-84].  Choose among:
.sp
.br
WGS-84 : World Geodetic System [Default] (1984)
.br
OSU91A : Ohio State University (1991)
.br
OSU86F : Ohio State University (1986)
.br
Engelis : Goddard Earth Models (1985)
.br
SGS-85 : Soviet Geodetic System (1985)
.br
TOPEX : Used commonly for altimetry (1990)
.br
MERIT-83 : United States Naval Observatory (1983)
.br
GRS-80 : International Geodetic Reference System (1980)
.br
Hughes-1980 : Hughes Aircraft Company for DMSP SSM/I grid products (1980)
.br
Lerch : For geoid modelling (1979)
.br
ATS77 : Average Terrestrial System, Canada Maritime provinces (1977)
.br
IAG-75 : International Association of Geodesy (1975)
.br
Indonesian : Applies to Indonesia (1974)
.br
WGS-72 : World Geodetic System (1972)
.br
NWL-10D : Naval Weapons Lab (Same as WGS-72) (1972)
.br
South-American : Applies to South America (1969)
.br
Fischer-1968 : Used by NASA for Mercury program (1968)
.br
Modified-Mercury-1968 : Same as Fischer-1968 (1968)
.br
GRS-67 : International Geodetic Reference System (1967)
.br
International-1967 : Worldwide use (1967)
.br
WGS-66 : World Geodetic System (1966)
.br
NWL-9D : Naval Weapons Lab (Same as WGS-66) (1966)
.br
Australian : Applies to Australia (1965)
.br
APL4.9 : Appl. Physics (1965)
.br
Kaula : From satellite tracking (1961)
.br
Hough : Applies to the Marshall Islands (1960)
.br
WGS-60 : World Geodetic System (1960)
.br
Fischer-1960 : Used by NASA for Mercury program (1960)
.br
Mercury-1960 : Same as Fischer-1960 (1960)
.br
Modified-Fischer-1960 : Applies to Singapore (1960)
.br
Fischer-1960-SouthAsia : Same as Modified-Fischer-1960 (1960)
.br
Krassovsky : Used in the (now former) Soviet Union (1940)
.br
War-Office : Developed by G. T. McCaw (1926)
.br
International-1924 : Worldwide use (1924)
.br
Hayford-1909 : Same as the International 1924 (1909)
.br
Helmert-1906 : Applies to Egypt (1906)
.br
Clarke-1880 : Applies to most of Africa, France (1880)
.br
Clarke-1880-Arc1950 : Modified Clarke-1880 for Arc 1950 (1880)
.br
Clarke-1880-IGN : Modified Clarke-1880 for IGN (1880)
.br
Clarke-1880-Jamaica : Modified Clarke-1880 for Jamaica (1880)
.br
Clarke-1880-Merchich : Modified Clarke-1880 for Merchich (1880)
.br
Clarke-1880-Palestine : Modified Clarke-1880 for Palestine (1880)
.br
Andrae : Applies to Denmark and Iceland (1876)
.br
Clarke-1866 : Applies to North America, the Philippines (1866)
.br
Clarke-1866-Michigan : Modified Clarke-1866 for Michigan (1866)
.br
Struve : Friedrich Georg Wilhelm Struve (1860)
.br
Clarke-1858 : Clarke's early ellipsoid (1858)
.br
Airy : Applies to Great Britain (1830)
.br
Airy-Ireland : Applies to Ireland in 1965 (1830)
.br
Modified-Airy : Same as Airy-Ireland (1830)
.br
Bessel : Applies to Central Europe, Chile, Indonesia (1841)
.br
Bessel-Schwazeck : Applies to Namibia (1841)
.br
Bessel-Namibia : Same as Bessel-Schwazeck (1841)
.br
Bessel-NGO1948 : Modified Bessel for NGO 1948 (1841)
.br
Everest-1830 : India, Burma, Pakistan, Afghanistan, Thailand (1830)
.br
Everest-1830-Kalianpur : Modified Everest for Kalianpur (1956) (1830)
.br
Everest-1830-Kertau : Modified Everest for Kertau, Malaysia & Singapore (1830)
.br
Everest-1830-Timbalai : Modified Everest for Timbalai, Sabah Sarawak (1830)
.br
Everest-1830-Pakistan : Modified Everest for Pakistan (1830)
.br
Walbeck : First least squares solution by Finnish astronomer (1819)
.br
Plessis : Old ellipsoid used in France (1817)
.br
Delambre : Applies to Belgium (1810)
.br
CPM : Comm. des Poids et Mesures, France (1799)
.br
Maupertius : Really old ellipsoid used in France (1738)
.br
Sphere : The mean radius in WGS-84 (for spherical/plate tectonics applications) (1984)
.br
Moon : Moon (IAU2000) (2000)
.br
Mercury : Mercury (IAU2000) (2000)
.br
Venus : Venus (IAU2000) (2000)
.br
Mars : Mars (IAU2000) (2000)
.br
Jupiter : Jupiter (IAU2000) (2000)
.br
Saturn : Saturn (IAU2000) (2000)
.br
Uranus : Uranus (IAU2000) (2000)
.br
Neptune : Neptune (IAU2000) (2000)
.br
Pluto : Pluto (IAU2000) (2000)
.sp
Note that for some global projections, \fBGMT\fP may use a spherical approximation of the
ellipsoid chosen, setting the flattening to zero, and using a mean radius.
A warning will be given when this happens.
If a different
ellipsoid name than those mentioned here is given, \fBGMT\fP will attempt to parse the
name to extract the semi-major axis (\fIa\fP in m) and the flattening. Formats allowed are:
.br
.sp
	\fIa\fP		implies a zero flattening
.br
	\fIa\fP,\fIinv_f\fP	where \fIinv_f\fP is the inverse flattening
.br
	\fIa\fP,\fBb=\fP\fIb\fP	where \fIb\fP is the semi-minor axis (in m)
.br
	\fIa\fP,\fBf=\fP\fIf\fP	where \fIf\fP is the flattening
.br
.sp
This way a
custom ellipsoid (e.g., those used for other planets) may be used. Further note that coordinate
transformations in \fBmapproject\fP can also specify specific datums; see the \fBmapproject\fP
man page for further details and how to view ellipsoid and datum parameters.
.TP
.B FIELD_DELIMITER
This setting determines what character will separate ASCII output data columns written by \fBGMT\fP.  Choose
from tab, space, comma, and none [tab].
.TP
.B FRAME_PEN
Pen attributes used to draw plain map frame in dpi units or points (append p)  [1.25p].
.TP
.B FRAME_WIDTH
Width (> 0) of map borders for fancy map frame [0.2\fBc\fP (or 0.075\fBi\fP)].
.TP
.B GLOBAL_X_SCALE
Global x-scale (> 0) to apply to plot-coordinates before plotting.  Normally used to shrink
the entire output down to fit a specific height/width  [1.0].
.TP
.B GLOBAL_Y_SCALE
Same, but for y-coordinates  [1.0].
.TP
.B GRID_CROSS_SIZE_PRIMARY
Size (>= 0) of grid cross at lon-lat intersections.  0 means draw continuous gridlines
instead [0].
.TP
.B GRID_CROSS_SIZE_SECONDARY
Size (>= 0) of grid cross at secondary lon-lat intersections.  0 means draw continuous gridlines
instead [0].
.TP
.B GRID_PEN_PRIMARY
Pen attributes used to draw grid lines in dpi units or points (append p) [0.25p].
.TP
.B GRID_PEN_SECONDARY
Pen attributes used to draw grid lines in dpi units or points (append p) [0.5p].
.TP
.B GRIDFILE_FORMAT
Default file format for grids, with optional scale, offset and invalid value, written as
\fIff\fP/\fIscale\fP/\fIoffset\fP/\fIinvalid\fP. The 2-letter format indicator can be one of [\fBbcnsr\fP][\fBbsifd\fP]. The first
letter indicates native \fBGMT\fP binary, old format netCDF, COARDS-compliant netCDF, Surfer format or Sun
Raster format. The second letter stands for byte, short, int, float and double, respectively. When
/\fIinvalid\fP is omitted the appropriate value for the given format is used (NaN or largest negative).
When /\fIscale\fP/\fIoffset\fP is omitted, /1.0/0.0 is used. [nf].
.TP
.B GRIDFILE_SHORTHAND
If TRUE, all grid file names are examined to see if they use the file extension
shorthand discussed in Section 4.17 of the \fBGMT\fP Technical Reference and Cookbook.
If FALSE, no filename expansion is done [FALSE].
.TP
.B HEADER_FONT
Font to use when plotting headers.  See \fBANNOT_FONT_PRIMARY\fP for available fonts  [Helvetica].
.TP
.B HEADER_FONT_SIZE
Font size (> 0) for header  [36p].
.TP
.B HEADER_OFFSET
Distance from top of axis annotations (or axis label, if present) to base of plot header [0.5\fBc\fP (or 0.1875\fBi\fP)].
.TP
.B HISTORY
If TRUE, passes the history of past common command options via the hidden .gmtcommands4 file [TRUE].
.TP
.B HSV_MAX_SATURATION
Maximum saturation (0\-1) assigned for most positive intensity value [0.1].
.TP
.B HSV_MIN_SATURATION
Minimum saturation (0\-1) assigned for most negative intensity value [1.0].
.TP
.B HSV_MAX_VALUE
Maximum value (0\-1) assigned for most positive intensity value [1.0].
.TP
.B HSV_MIN_VALUE
Minimum value (0\-1) assigned for most negative intensity value [0.3].
.TP
.B INPUT_CLOCK_FORMAT
Formatting template that indicates how an input clock string is formatted.  This template
is then used to guide the reading of clock strings in data fields.  To properly decode
12-hour clocks, append am or pm (or upper case) to match your data records. As examples,
try hh:mm, hh:mm:ssAM, etc. [hh:mm:ss].
.TP
.B INPUT_DATE_FORMAT
Formatting template that indicates how an input date string is formatted.  This template
is then used to guide the reading of date strings in data fields.  You may specify either
Gregorian calendar format or ISO week calendar format.  Gregorian calendar: Use any
combination of yyyy (or yy for 2-digit years; if so see \fBY2K_OFFSET_YEAR\fP), mm (or o
for abbreviated month name in the current time language),
and dd, with or without delimiters.  For day-of-year data, use jjj instead of mm and/or dd.
Examples can be ddmmyyyy, yy-mm-dd, dd-o-yyyy, yyyy/dd/mm, yyyy-jjj, etc.
ISO Calendar:  Expected template is yyyy[-]W[-]ww[-]d, where ww is ISO week and d is ISO week
day.  Either template must be consistent, e.g., you cannot specify months if you don't specify\"'
years.  Examples are yyyyWwwd, yyyy-Www, etc. [yyyy-mm-dd].
.TP
.B INTERPOLANT
Determines if linear (linear), Akima's spline (akima), natural cubic spline (cubic) or\"'
no interpolation (none) should be used for 1-D interpolations in various programs [akima].
.TP
.B IO_HEADER
(\fB* \-H\fP) Specifies whether input/output ASCII files have header record(s) or not  [FALSE].
.TP
.B LABEL_FONT
Font to use when plotting labels below axes.  See \fBANNOT_FONT_PRIMARY\fP for available fonts  [Helvetica].
.TP
.B LABEL_FONT_SIZE
Font size (> 0) for labels  [24p].
.TP
.B LABEL_OFFSET
Distance from base of axis annotations to the top of the axis label [0.3\fBc\fP (or 0.1125\fBi\fP)].
.TP
.B LINE_STEP
Determines the maximum length (> 0) of individual straight line-segments when drawing arcuate lines  [0.025\fBc\fP (or 0.01\fBi\fP)]
.TP
.B MAP_SCALE_FACTOR
Changes the default map scale factor used for the Polar Stereographic [0.9996], UTM [0.9996], and Transverse Mercator [1] projections
in order to minimize areal distortion.  Provide a new scale-factor or leave as default.
.TP
.B MAP_SCALE_HEIGHT
Sets the height (> 0) on the map of the map scale bars drawn by various programs [0.2\fBc\fP (or 0.075\fBi\fP)].
.TP
.B MEASURE_UNIT
Sets the unit length.  Choose between cm, inch, m, and point.  [cm].  Note
that, in \fBGMT\fP, one point is defined as 1/72 inch (the \fIPostScript\fP definition), while it
is often defined as 1/72.27 inch in the typesetting industry.  There is no
universal definition.
.TP
.B N_COPIES
(\fB* \-c\fP) Number of plot copies to make  [1].
.TP
.B N_HEADER_RECS
Specifies how many header records to expect if \fB\-H\fP is turned on  [1].
.TP
.B NAN_RECORDS
Determines what happens when input records containing NaNs for \fIx\fP or \fIy\fP (and in some
cases \fIz\fP) are read.  Choose between \fIskip\fP, which will simply report how many bad
records were skipped, and \fIpass\fP [Default], which will pass these records on to the calling programs.
For most programs this will result in output records with NaNs as well, but some will interpret these
NaN records to indicate gaps in a series; programs may then use that information to detect segmentation (if
applicable).
.TP
.B OBLIQUE_ANNOTATION

ChangeLog
-----------------------------------------------------------------------------------------
	$Id: ChangeLog 12925 2014-02-20 22:02:24Z pwessel $
-----------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------
201x-yy-zz	pwessel
	GMT 5.1.1 finalized and released.
-----------------------------------------------------------------------------------------
2014-02-20	pwessel
	* grdcontour.c:		The -Zscale/offset option was applied incorrectly due to a
				change in the underlying scaling function.
	* grdvolume.c:		---"---
2014-02-10	pwessel
	*gmt2kml.c:		Make sure the default style IDs are unique for each process.

2014-02-03	pwessel
	* pslib.c:		Auto-widen paragraph width to at least fit widest word.

2014-01-26	pwessel
	* gmt_grdio.c:		Building the grid command string could exceed its length if
				the argument became exactly the max length of the array; then
				a final extra character \0 would be added and exceed array length.
				Thanks to Joachim Saul for pointing it out.

2014-01-22	pwessel
	* pscoast.c:		The processing of -I -N levels and pens could get out of sync
				so that the wrong pen was used for the specified feature.
2014-01-16	remko
	* pslib.c:		Take superscripts and subscripts into account when determining
				dimensions of text box.

2014-01-10	pwessel
	* grdmath.c:		We consider the change in SDIST, PDIST to return degrees instead
				of km to be a bug as it makes it difficult to handle geodesics
				for which degrees make no sense.  Thus we have consolidated all
				the ?DIST operators to give km if geographic data are involved.
				Example 23 has been adjusted accordingly, as has several test
				scripts.  Please check your own scripts for SDIST.

2014-01-09	pwessel
	* gmt_vector.c:		Add protection against introduced longitude-jumps produced
				by resampling in GMT_fix_up_path.
2014-01-09	remko
	* gmt_vector.c:		Add protection against introduced longitude-jumps produced
				by resampling in GMT_fix_up_path.
	* grdmath.c:		LDIST and PDIST will again produce km when using geodesics.

2014-01-01	pwessel
	* x2sys/x2sys.c:	Mishandled the assignment of segment number for each record.
	* gmt_support.c:	Allow GMT_crossover to deal with global wrapped data.

2013-12-31	pwessel
	* gmt2kml.c:		Now using GMT_Put_Record instead of old stdout printing.
				This addresses issue #473.

2013-12-10	pwessel
	* grdvector.c:		Vector direction did not adjust when negative scales were
				used in -JX or -Jx.
	* psxy[z].c		---"---

2013-12-08	remko
	* psimage.c:		Now properly plots transparent images and makes -Gt work.

2013-11-26	pwessel
	* gmt_io.c:		GMT would crash on a blank line with no newline at end at all.
	* pstext.c:		pstext did not process GMT4-style input properly.

2013-11-26	jluis
	* potential/gravfft.c:	Let -E options default to 3.
				Let users select the -N+l (leave trend alone). That is, do not overwrite with default mode.
				Let the default detrend be remove MID point.
				Explicitly set -N+a in gravfft command to avoid failures due to coming changes in gravfft.
2013-11-25	pwessel
	* grdlandmask.c:	Failed to fill partial bins with no coastlines when the region
				crossed Greenwich [issue #451].

2013-11-21	pwessel
	* pscontour.c:		Let pscontour output contours to file [issue #448].
	* pscoast.c:		Prevent users from not giving proper projection [issue #449].

2013-11-19	jluis
	* mgd77/mgd77magref.c:	Needed to reset the t_col variable when constant altitude [issue #446].

2013-11-19	pwessel
	* gmt_sph.c:		Fix invalid read in ssrfpack_grid due to par array being of length 2 rather than 3.

2013-11-18	pwessel
	* gmt_grdio.c:		Place the old auto-detect of 4-byte binary float vs int under compatibility mode for backwardness.

2013-11-16	pwessel
	* grdvector.c:		Address shortcomings of geovectors when wrapped and clipped.

2013-11-15	pwessel
	* geovectors.c:		Fix geovector when shrinking is selected.

2013-11-13	pwessel
	* gmt_api.c:		Fix API issues pointed out by Ian.

2013-11-13	fwobbe
	* GMT.app:		Inherit user shell in GMT.app.

2013-11-12	pwessel
	* gmt_init.c:		grdmath could corrupt the gmt.history file [issue # 438].
 	
2013-11-10	pwessel
	* gmt_support.c:	Fix symbol-size issue in psxy -Sc [issue #435].
	
2013-11-09	pwessel
	* grdvector.c:		Did not convert direction to azimuths for geo vectors.

2013-11-08	pwessel
	* grdvector.c:		Fix grdvector -I problem [issue #434].
	* ps2raster.c:		Under Mavericks, strcpy would exit if the to and from pointers
-----------------------------------------------------------------------------------------
2013-11-05	pwessel
	GMT 5.1.0 finalized and released.
-----------------------------------------------------------------------------------------

2013-11-01	remko
	* example_10.sh:	Changed input to number of languages; multi-color columns

2013-10-31	pwessel
	* gmt_dcw.c:		Added option to specify code as =<continent> which will
				automatically be expanded to all the countries in that
				continent.  E.g., -F=AF+gred will paint Africa read.
				Fixed memory leaks.

2013-10-23	pwessel
	* gmtinfo.c:		The new name of minmax.c, the latter will be honored
				when GMT compatibility mode is 4.
	* grdmath.c:		Fixed porting bugs for CURV and D2DY2; added test ops.sh

2013-10-22	pwessel
	* minmax.c:		Added -D to adjust region found with -I
	* makecpt.c:		Better reporting of gaps or overlaps in CPTs

2013-10-05	pwessel
	* greenspline.c:	Needed -f in order to select -f0T input, plus it made assumptions
				about lon,lat despite not being selected.

2013-10-04	pwessel
	* gmt.history:		This replaces the old .gmtcommands file name.

2013-09-28	remko
	* gmt_init.c:		Moved origin of JD by one day (24 Nov instead of 25 Nov).
	* ConfigUserTemplate.cmake:	Added option GMT_DOCS_DEPEND_ON_GMT

2013-09-27	pwessel
	* share/cpt:		Added 10 new CPTs, mostly for bathymetry and topography.

2013-09-14	pwessel

src/gmtdefaults.c
/*--------------------------------------------------------------------
 *	$Id: gmtdefaults.c 12822 2014-01-31 23:39:56Z remko $
 *
 *	Copyright (c) 1991-2014 by P. Wessel, W. H. F. Smith, R. Scharroo, J. Luis and F. Wobbe
 *	See LICENSE.TXT file for copying and redistribution conditions.
 *
 *	This program is free software; you can redistribute it and/or modify
 *	it under the terms of the GNU Lesser General Public License as published by
 *	the Free Software Foundation; version 3 or any later version.
 *
 *	This program is distributed in the hope that it will be useful,
 *	but WITHOUT ANY WARRANTY; without even the implied warranty of
 *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *	GNU Lesser General Public License for more details.
 *
 *	Contact info: gmt.soest.hawaii.edu
 *--------------------------------------------------------------------*/
/*
 * Author:	Paul Wessel
 * Date:	1-JAN-2010
 * Version:	5 API
 *
 * Brief synopsis: gmtdefaults will list the user's GMT default settings
 * or (by using the -D option), get the site GMT default settings.
 *
 */
 
#define THIS_MODULE_NAME	"gmtdefaults"
#define THIS_MODULE_LIB		"core"
#define THIS_MODULE_PURPOSE	"List current GMT default parameters"

#include "gmt_dev.h"

#define GMT_PROG_OPTIONS "-V"

/* Control structure for gmtdefaults */

struct GMTDEFAULTS_CTRL {
	struct D {	/* -D[s|u] */
		bool active;
		char mode;
	} D;
};

void *New_gmtdefaults_Ctrl (struct GMT_CTRL *GMT) {	/* Allocate and initialize a new control structure */
	struct GMTDEFAULTS_CTRL *C;

	C = GMT_memory (GMT, NULL, 1, struct GMTDEFAULTS_CTRL);
	return (C);
}

void Free_gmtdefaults_Ctrl (struct GMT_CTRL *GMT, struct GMTDEFAULTS_CTRL *C) {	/* Deallocate control structure */
	if (!C) return;
	GMT_free (GMT, C);	
}

int GMT_gmtdefaults_usage (struct GMTAPI_CTRL *API, int level)
{
	GMT_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_NAME, THIS_MODULE_PURPOSE);
	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
	GMT_Message (API, GMT_TIME_NONE, "usage: gmtdefaults [-D[s|u]]\n\n");
	
	if (level == GMT_SYNOPSIS) return (EXIT_FAILURE);
	
	GMT_Message (API, GMT_TIME_NONE, "\t-D Print the GMT default settings.\n");
	GMT_Message (API, GMT_TIME_NONE, "\t   Append s to see the SI version of defaults.\n");
	GMT_Message (API, GMT_TIME_NONE, "\t   Append u to see the US version of defaults.\n");
	
	return (EXIT_FAILURE);
}

int GMT_gmtdefaults_parse (struct GMT_CTRL *GMT, struct GMTDEFAULTS_CTRL *Ctrl, struct GMT_OPTION *options)
{
	/* This parses the options provided to gmtdefaults and sets parameters in CTRL.
	 * Any GMT common options will override values set previously by other commands.
	 * It also replaces any file names specified as input or output with the data ID
	 * returned when registering these sources/destinations with the API.
	 */

	unsigned int n_errors = 0, n_files = 0;
	struct GMT_OPTION *opt = NULL;

	for (opt = options; opt; opt = opt->next) {
		switch (opt->option) {

			case '<':	/* Count input files */
				n_files++;
				break;

			/* Processes program-specific parameters */

			case 'D':	/* Get GMT system-wide defaults settings */
				Ctrl->D.active = true;
				Ctrl->D.mode = opt->arg[0];
				break;
			case 'L':	/* List the user's current GMT defaults settings */
				if (GMT_compat_check (GMT, 4)) {
					GMT_Report (GMT->parent, GMT_MSG_COMPAT, "Warning: Option -L is deprecated; it is now the default behavior.\n");
				}
				else
					n_errors += GMT_default_error (GMT, opt->option);
				break;

			default:	/* Report bad options */
				n_errors += GMT_default_error (GMT, opt->option);
				break;
		}
	}

	n_errors += GMT_check_condition (GMT, n_files, "Syntax error: No input files are expected\n");

	return (n_errors ? GMT_PARSE_ERROR : GMT_OK);
}

/* Must free allocated memory before returning */
#define bailout(code) {GMT_Free_Options (mode); return (code);}
#define Return(code) {Free_gmtdefaults_Ctrl (GMT, Ctrl); GMT_end_module (GMT, GMT_cpy); bailout (code);}

int GMT_gmtdefaults (void *V_API, int mode, void *args)
{
	int error;
	
	char path[GMT_LEN256] = {""};
	
	struct GMTDEFAULTS_CTRL *Ctrl = NULL;
	struct GMT_CTRL *GMT = NULL, *GMT_cpy = NULL;
	struct GMT_OPTION *options = NULL;
	struct GMTAPI_CTRL *API = GMT_get_API_ptr (V_API);	/* Cast from void to GMTAPI_CTRL pointer */

	/*----------------------- Standard module initialization and parsing ----------------------*/

	if (API == NULL) return (GMT_NOT_A_SESSION);
	if (mode == GMT_MODULE_PURPOSE) return (GMT_gmtdefaults_usage (API, GMT_MODULE_PURPOSE));	/* Return the purpose of program */
	options = GMT_Create_Options (API, mode, args);	if (API->error) return (API->error);	/* Set or get option list */

	if (options) {
		if (options->option == GMT_OPT_USAGE) bailout (GMT_gmtdefaults_usage (API, GMT_USAGE));		/* Return the usage message */
		if (options->option == GMT_OPT_SYNOPSIS) bailout (GMT_gmtdefaults_usage (API, GMT_SYNOPSIS));	/* Return the synopsis */
	}

	/* Parse the command-line arguments */

	GMT = GMT_begin_module (API, THIS_MODULE_LIB, THIS_MODULE_NAME, &GMT_cpy); /* Save current state */
	if (GMT_Parse_Common (API, GMT_PROG_OPTIONS, options)) Return (API->error);
	Ctrl = New_gmtdefaults_Ctrl (GMT);	/* Allocate and initialize a new control structure */
	if ((error = GMT_gmtdefaults_parse (GMT, Ctrl, options))) Return (error);

	/*---------------------------- This is the gmtdefaults main code ----------------------------*/

	if (Ctrl->D.active) {
		GMT_getsharepath (GMT, "conf", "gmt", (Ctrl->D.mode == 's') ? "_SI" : (Ctrl->D.mode == 'u') ? "_US" : ".conf", path, R_OK);
		GMT_loaddefaults (GMT, path);
	}

	GMT_putdefaults (GMT, "-");

	Return (GMT_OK);
}
